tensorflow/core/grappler/optimizers/auto_mixed_precision_test.cc

```diff
@@ -111,7 +111,7 @@ class AutoMixedPrecisionTest : public GrapplerTest {
 #endif
     if (gpu_available_) {
       virtual_cluster_.reset(new SingleMachine(/* timeout_s = */ 10, 1, 1));
-    } else {
+    } else if( num_gpus > 0) {
```

Comment:
We should still run GPU tests with the fake GPUs even when there are no GPUs available.

Use the value of `GetParam()` to determine if we're testing CPUs or GPUs. For GPUs, use the original logic. For CPUs, use the new logic where you use a CPU.

Reply:
done